### PR TITLE
DOC: Clean up method Doxygen documentation

### PR DESCRIPTION
**Pull request by jhlegarreta**
Clean up method Doxygen documentation:
- Consolidate documentation so that methods are documented in the header
  files. By doing this fix the documentation of methods which had the
  wrong documentation (e.g. copied from another existing, unrelated
  method).
- Remove the documentation of trivial methods: constructors,
  destructors, `PrintSelf` methods, etc.
- Remove non-descriptive documentation of instance variables whose
  documentation should dwell in their public Get/Set macros according to
  the ITK SWG coding style guideline.
- Remove documentation blocks consisting solely of the method name from
  the implementation files.
- Remove unnecessary method ending brace comment blocks.
- Remove empty method documentation blocks from implementation file.
- Fix typos and method names cross-referenced in the documentation.
- Take advantage of the commit to improve the documentation style
  (capitalize first letter in sentence, finish sentences with a period,
  split brief and detailed documentation sentences, etc.)

Note that many of the implementation file method documentation blocks were not in a Doxygen-recognizable format either:
https://doxygen.nl/manual/docblocks.html

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [X] Updated API documentation (or API not changed)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
This is by no means intended to be exhaustive. I was having a look at some other things across these files, and thought that I would take the time to the consolidate documentation in these files, clean up unnecessary blocks, etc.

File: Modules/Core/ImageFunction/include/itkCentralDifferenceImageFunction.h [hunk 1]
```diff
@@ -217,37 +217,43 @@ class ITK_TEMPLATE_EXPORT CentralDifferenceImageFunction : public ImageFunction<
...
+
+  /** Specialized version of EvaluateAtContinuousIndex() method to handle vector pixel types.*/
   template <typename Type>
   inline void
   EvaluateAtContinuousIndexSpecialized(const ContinuousIndexType & cindex,
                                        OutputType &                derivative,
                                        OutputTypeSpecializationStructType<Type>) const;
 
-  /** Specialized versions of Evaluate() method to handle scalar or vector pixel types.*/
+  /** Specialized version of Evaluate() method to handle scalar pixel types.*/
```
Comment (thewtex) [inline Modules/Core/ImageFunction/include/itkCentralDifferenceImageFunction.h:248]: Is the removal of `or vector` intended?

Comment (jhlegarreta) [reply Modules/Core/ImageFunction/include/itkCentralDifferenceImageFunction.h:248]: Yes Matt. It looks like there are two versions for each of these methods, one for scalar types and the other one for vector types; at least that was what was being said in the implementation file. Please check.

**Comment by jhlegarreta** [comment]
There are surely a few hundreds of other files where this should be done, but they can be addressed in separate PRs as time permits. To me, this is a step forward.

**Review by dzenanz — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Pull request merged**